skimage/measure/tests/test_regionprops.py

```diff
@@ -423,104 +434,21 @@ def test_invalid_size():
     with testing.raises(ValueError):
         regionprops(SAMPLE, wrong_intensity_sample)
 
-
-def test_equals():
-    arr = np.zeros((100, 100), dtype=np.int)
-    arr[0:25, 0:25] = 1
-    arr[50:99, 50:99] = 2
-
-    regions = regionprops(arr)
-    r1 = regions[0]
-
-    regions = regionprops(arr)
-    r2 = regions[0]
-    r3 = regions[1]
-
-    assert_equal(r1 == r2, True, "Same regionprops are not equal")
-    assert_equal(r1 != r3, True, "Different regionprops are equal")
-
-
-def test_iterate_all_props():
-    region = regionprops(SAMPLE)[0]
-    p0 = {p: region[p] for p in region}
-
-    region = regionprops(SAMPLE, intensity_image=INTENSITY_SAMPLE)[0]
-    p1 = {p: region[p] for p in region}
-
-    assert len(p0) < len(p1)
-
-
-def test_cache():
-    region = regionprops(SAMPLE)[0]
-    f0 = region.filled_image
-    region._label_image[:10] = 1
-    f1 = region.filled_image
-
-    # Changed underlying image, but cache keeps result the same
-    assert_array_equal(f0, f1)
-
-    # Now invalidate cache
-    region._cache_active = False
-    f1 = region.filled_image
-
-    assert np.any(f0 != f1)
-
-
-def test_docstrings_and_props():
-    def foo():
-        """foo"""
-
-    has_docstrings = bool(foo.__doc__)
-
-    region = regionprops(SAMPLE)[0]
-
-    docs = _parse_docs()
-    props = [m for m in dir(region) if not m.startswith('_')]
-
-    nr_docs_parsed = len(docs)
-    nr_props = len(props)
-    if has_docstrings:
-        assert_equal(nr_docs_parsed, nr_props)
-        ds = docs['weighted_moments_normalized']
-        assert 'iteration' not in ds
-        assert len(ds.split('\n')) > 3
-    else:
-        assert_equal(nr_docs_parsed, 0)
-
-
-def test_props_to_dict():
-    regions = regionprops(SAMPLE)
-    out = _props_to_dict(regions)
-    assert out == {'label': array([1]),
-                   'bbox-0': array([0]), 'bbox-1': array([0]),
-                   'bbox-2': array([10]), 'bbox-3': array([18])}
-
-    regions = regionprops(SAMPLE)
-    out = _props_to_dict(regions, properties=('label', 'area', 'bbox'),
-                         separator='+')
-    assert out == {'label': array([1]), 'area': array([180]),
-                   'bbox+0': array([0]), 'bbox+1': array([0]),
-                   'bbox+2': array([10]), 'bbox+3': array([18])}
-
-
-def test_regionprops_table():
-    out = regionprops_table(SAMPLE)
-    assert out == {'label': array([1]),
-                   'bbox-0': array([0]), 'bbox-1': array([0]),
-                   'bbox-2': array([10]), 'bbox-3': array([18])}
-
-    out = regionprops_table(SAMPLE, properties=('label', 'area', 'bbox'),
```

Comment:
Why are these tests removed?

Reply:
Ah, that was my mistake. Messed up when merging and lost the recently added code. I have now added all these new tests back (except for old test of parsing of the documentation). All tests now passing.

In doing so, I found a nasty bug. I noticed in the new testers for the regionprops_table and dict that it was asserting the area of the SAMPLE global as 180. This is clearly incorrect and the area should be 72. It turns out this was happening because the old cache tester (which I have replaced) had a bug which swapped out the SAMPLE global (to all ones) when that test was run. So I guess a lesson to be sure to check the actual values independently when making unit tests.

Reply:
Thats really funny, i JUST made a PR for that bug

https://github.com/scikit-image/scikit-image/pull/4099

Reply:
Thats really funny, i JUST made a PR for that bug

https://github.com/scikit-image/scikit-image/pull/4099

Reply:
I'm really sorry that there are 2 PRs in flight that are going to caues you merge issues......

Reply:
Hah. So I guess I should check the other pull requests frequently. Would have saved me quite a while figuring out what the heck was going on with that bug.

Maybe it would be a good idea to use some sort of immutable container like frozendict for the test data to make sure that this kind of thing can't happen.

Reply:
code inertia is real. Unless there is a real problem with the functionality code (python optimize for example), probably best not to touch it. I was just running the test suite alot, and these bugs were causing parallel tests to fail.

This code was also probably written before the days of the property decorator, and the popularity of these cool containers.